torch/_dynamo/source.py

```diff
@@ -449,25 +449,7 @@ def guard_source(self):
 
 
 @dataclasses.dataclass(frozen=True)
-class DeterministicAlgorithmsSource(Source):
-    def name(self):
-        return ""
-
-    def guard_source(self):
-        return GuardSource.GLOBAL
-
-
-@dataclasses.dataclass(frozen=True)
-class GradModeSource(Source):
-    def name(self):
-        return ""
-
-    def guard_source(self):
-        return GuardSource.GLOBAL
-
-
-@dataclasses.dataclass(frozen=True)
-class DefaultDeviceSource(Source):
+class GlobalStateSource(Source):
     def name(self):
```

Comment:
should this raise? when is named used or invoked?

Reply:
https://github.com/pytorch/pytorch/blob/5b4aacd691725ce433954aef5101d12c53d88af8/torch/_guards.py#L587

Reply:
Discussing offline. Unclear if this is what we want, but also not fair to you to ask to do this on this PR. I will refactor this myself later.